packages/universal-components/src/RangeDatePicker/AnimatedDayItemArrow.js

```diff
@@ -34,35 +34,56 @@ const RenderArrows = ({ scale }) => (
 );
 
 export default class AnimatedDayItemArrow extends React.Component<Props> {
-  static scale: Animated.Value = new Animated.Value(0);
+  static scale: Animated.Value;
   static animating: boolean;
 
+  static numberOfInstances: number;
+
   static handleAnimation() {
-    AnimatedDayItemArrow.scale.setValue(0);
+    if (AnimatedDayItemArrow.scale === undefined) {
+      AnimatedDayItemArrow.scale = new Animated.Value(0);
+    }
 
-    Animated.sequence([
-      Animated.timing(AnimatedDayItemArrow.scale, {
-        ...SHARED_ANIMATION_CONFIG,
-        toValue: 1,
-      }),
-      Animated.timing(AnimatedDayItemArrow.scale, {
-        ...SHARED_ANIMATION_CONFIG,
-        toValue: 0,
-      }),
-    ]).start(() => {
-      if (AnimatedDayItemArrow.animating) {
-        AnimatedDayItemArrow.handleAnimation();
-      }
-    });
+    if (AnimatedDayItemArrow.scale._value === 0) {
+      Animated.sequence([
+        Animated.timing(AnimatedDayItemArrow.scale, {
+          ...SHARED_ANIMATION_CONFIG,
+          toValue: 1,
+        }),
+        Animated.timing(AnimatedDayItemArrow.scale, {
+          ...SHARED_ANIMATION_CONFIG,
+          toValue: 0.1,
+        }),
+      ]).start(() => {
+        AnimatedDayItemArrow.scale.setValue(0);
+        if (AnimatedDayItemArrow.animating) {
+          AnimatedDayItemArrow.handleAnimation();
+        }
+      });
+    }
   }
 
   componentDidMount() {
-    AnimatedDayItemArrow.animating = true;
-    AnimatedDayItemArrow.handleAnimation();
+    if (AnimatedDayItemArrow.numberOfInstances === undefined) {
```

Comment:
Definitely looks better now. Can't say I like all those conditions, but I can probably live with it if it's necessary for synchronisation 😄  .

Just `ios` version still acts weirdly. If I open picker for second time, animation goes to 100% then freezes for about second and then continues as expected.

![arrow freeze](https://user-images.githubusercontent.com/2660330/56289248-cac0ee00-6120-11e9-9b7a-73761f20871b.gif)

Reply:
OK, should be fixed. Animation start is now wrapped in `InteractionManager.runAfterInteractions`

Reply:
Looks good now! Thanks